packages/language-server/lib/convert.js

```diff
@@ -0,0 +1,251 @@
+/**
+ * @typedef {import('typescript')} ts
+ * @typedef {import('typescript').CompletionEntryDetails} CompletionEntryDetails
+ * @typedef {import('typescript').DefinitionInfo} DefinitionInfo
+ * @typedef {import('typescript').Diagnostic} Diagnostic
+ * @typedef {import('typescript').DiagnosticCategory} DiagnosticCategory
+ * @typedef {import('typescript').DiagnosticMessageChain} DiagnosticMessageChain
+ * @typedef {import('typescript').DiagnosticRelatedInformation} DiagnosticRelatedInformation
+ * @typedef {import('typescript').JSDocTagInfo} JSDocTagInfo
+ * @typedef {import('typescript').QuickInfo} QuickInfo
+ * @typedef {import('typescript').SymbolDisplayPart} SymbolDisplayPart
+ * @typedef {import('typescript').ScriptElementKind} ScriptElementKind
+ * @typedef {import('typescript').TextSpan} TextSpan
+ * @typedef {import('vscode-languageserver').Diagnostic} LspDiagnostic
+ * @typedef {import('vscode-languageserver').DiagnosticRelatedInformation} LspDiagnosticRelatedInformation
+ * @typedef {import('vscode-languageserver-textdocument').TextDocument} TextDocument
+ */
+
+import {
+  CompletionItemKind,
+  DiagnosticSeverity,
+  DiagnosticTag,
+  LocationLink,
+  Range,
+} from 'vscode-languageserver'
+
+import { documents } from './documents.js'
+
+/**
+ * @param {ScriptElementKind} kind
+ * @returns {CompletionItemKind} The matching Monaco completion item kind.
+ */
+export function convertScriptElementKind(kind) {
+  switch (kind) {
+    case 'primitive type':
+    case 'keyword':
+      return CompletionItemKind.Keyword
+    case 'var':
+    case 'local var':
+      return CompletionItemKind.Variable
+    case 'property':
+    case 'getter':
+    case 'setter':
+      return CompletionItemKind.Field
+    case 'function':
+    case 'method':
+    case 'construct':
+    case 'call':
+    case 'index':
+      return CompletionItemKind.Function
+    case 'enum':
+      return CompletionItemKind.Enum
+    case 'module':
+      return CompletionItemKind.Module
+    case 'class':
+      return CompletionItemKind.Class
+    case 'interface':
+      return CompletionItemKind.Interface
+    case 'warning':
+      return CompletionItemKind.File
+  }
+  return CompletionItemKind.Property
+}
+
+/**
+ * @param {ts} ts
+ * @param {CompletionEntryDetails | QuickInfo} details
+ * @returns {string} XXX
```

Comment:
Is this a placeholder for future documentation? Or is there an issue with the return?

Reply:
This is a placeholder. As code was changing fast, it was tedious to document everything.

Although switching from the current ESLint preset to XO would allow us to remove it, which I think is also fine for internals.